dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLMemoryStore.cs

```diff
@@ -164,6 +164,12 @@ await this._cosmosClient
         MemoryRecord record,
         CancellationToken cancellationToken = default)
     {
+        // In some cases we're expected to generate the key to use. Do so if one isn't provided.
+        if (string.IsNullOrEmpty(record.Key))
+        {
+            record.Key = Guid.NewGuid().ToString();
```

Comment:
Is this valid to do? We're modifying the input param here.

Reply:
This is temporary solution which exists in other memory connectors as well. We will remove it as soon as new memory abstraction will be in place.